### PR TITLE
Use golden package for unit test 🐐

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This allow to remove the potential long expected output from the test
code. See gotest.tools/v3/golden for documentation.

/cc @chmouel 

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Regenerate the manpages and docs with `make docs` and `make man` if needed.
- [x] Run the code checkers with `make check`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by chmouel** [comment]
What about a script or makefile target that would generate those golden files?

**Comment by vdemeester** [comment]
> What about a script or makefile target that would generate those golden files?

Yeah it's the next step :stuck_out_tongue_closed_eyes:

File: hack/update-golden.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env bash
+go test $(go list -f '{{ .ImportPath }} {{ .TestImports }}' ./... | grep golden | awk '{print $1}' | tr '\n' ' ') -test.update-golden=true
```
Comment (chmouel) [inline hack/update-golden.sh]: can you add a comment on what it does? this look like hebrewcr to me! (

Comment (vdemeester) [reply hack/update-golden.sh]: lol ok :sweat_smile:

**Comment by chmouel** [comment]
/lgtm

Looking Good!

**Pull request merged**